### PR TITLE
Add $connectionStatus tests

### PR DESCRIPTION
**Pull request by PatersonProjects**
This PR adds the tests for validating the $connectionStatus diagnostic command, primarily through type checking of response parameters.

Ref: Issue #43

### CODE REVIEW

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_argument_validation.py [hunk 1]
```diff
@@ -0,0 +1,58 @@
+"""Tests for connectionStatus command argument validation.
+
+Verifies that connectionStatus accepts all 18 BSON types (int, double, long,
+decimal128, string, bool, date, null, object, array, binData, objectId, regex,
+timestamp, minKey, maxKey) as the command field value and returns ok: 1 for each.
+"""
+
+from __future__ import annotations
```
Comment (vic-tsang) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_argument_validation.py]: I think this might not be necessary. Please remove as we are running from python 3.9+

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_show_privileges.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
+"""Tests for connectionStatus showPrivileges parameter.
+
+Verifies the truthy/falsy/omit behavior of the showPrivileges field.
+Truthy values (true, int 1, double 1.0, long 1, decimal128 1) should cause
+authenticatedUserPrivileges to appear as an array. Falsy values (false,
+int 0, double 0.0, long 0, decimal128 0, null) and omitting the field
+entirely should exclude authenticatedUserPrivileges from the response.
+"""
+
+from __future__ import annotations
```
Comment (vic-tsang) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_show_privileges.py]: please remove see above comment

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_argument_validation.py [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+    ),
+    ConnStatusArgTest("null", value=None, msg="null should succeed"),
+    ConnStatusArgTest("object", value={}, msg="object should succeed"),
+    ConnStatusArgTest("array", value=[], msg="array should succeed"),
+    ConnStatusArgTest("binData", value=Binary(b""), msg="binData should succeed"),
+    ConnStatusArgTest("objectId", value=ObjectId(), msg="objectId should succeed"),
+    ConnStatusArgTest("regex", value=Regex("test"), msg="regex should succeed"),
+    ConnStatusArgTest("timestamp", value=Timestamp(0, 0), msg="timestamp should succeed"),
+    ConnStatusArgTest("minKey", value=MinKey(), msg="minKey should succeed"),
+    ConnStatusArgTest("maxKey", value=MaxKey(), msg="maxKey should succeed"),
```
Comment (vic-tsang) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_argument_validation.py]: missing javascript Code

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_show_privileges.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+TRUTHY_TESTS: list[ShowPrivTest] = [
+    ShowPrivTest("true", show_priv=True, msg="true should show privileges"),
+    ShowPrivTest("int_1", show_priv=1, msg="int 1 (truthy) should show privileges"),
+    ShowPrivTest("double_1", show_priv=1.0, msg="double 1.0 (truthy) should show"),
+    ShowPrivTest("long_1", show_priv=Int64(1), msg="long 1 (truthy) should show"),
+    ShowPrivTest("decimal128_1", show_priv=Decimal128("1"), msg="decimal128 1 should show"),
+]
+
+FALSY_TESTS: list[ShowPrivTest] = [
+    ShowPrivTest("false", show_priv=False, msg="false should hide privileges"),
```
Comment (vic-tsang) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_show_privileges.py]: Please add missing test case negative value -1 , -1.0

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_errors.py [hunk 5]
```diff
@@ -0,0 +1,103 @@
...
+        error_code=TYPE_MISMATCH_ERROR,
+        msg="showPrivileges objectId should be rejected as wrong type",
+    ),
+    DiagnosticErrorTest(
+        id="showPrivileges_minKey_type_mismatch",
+        command={"connectionStatus": 1, "showPrivileges": MinKey()},
+        error_code=TYPE_MISMATCH_ERROR,
+        msg="showPrivileges minKey should be rejected as wrong type",
+    ),
+    DiagnosticErrorTest(
```
Comment (vic-tsang) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_errors.py]: missing binData, regex, timestamp, and javascript code

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_argument_validation.py [hunk 6]
```diff
@@ -0,0 +1,58 @@
+"""Tests for connectionStatus command argument validation.
+
+Verifies that connectionStatus accepts all 18 BSON types (int, double, long,
```
Comment (vic-tsang) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_argument_validation.py]: as some of the BSON type are deprecated, we might not be testing all 18 BSON types. Please update this doc string

**Review by vic-tsang — CHANGES_REQUESTED** [review]

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_response_validation.py [hunk 7]
```diff
@@ -0,0 +1,91 @@
...
+@pytest.mark.parametrize("test", pytest_params(RESPONSE_PROPERTY_TESTS))
+def test_connectionStatus_response_properties(collection, test):
+    """Verify a response field exists and has the expected type."""
+    result = execute_admin_command(collection, {"connectionStatus": 1})
+    assertProperties(result, test.checks, msg=test.msg, raw_res=True)
+
+
+def test_connectionStatus_succeeds_on_nonexistent_database(collection):
+    """Verify connectionStatus returns ok and authInfo when run on a non-existent database."""
+    other_col = collection.database.client["nonexistent_db_for_connstatus_test"]["dummy"]
```
Comment (eerxuan) [inline documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_response_validation.py]: Please don't use hard coded db name or collection name, use fixture-derived names (collection.name_{some_suffix}) to avoid cleanup leaks and parallel-run conflicts. Same for test_connectionStatus_same_result_across_databases.

**Review by eerxuan — COMMENTED** [review]
Please remove hardcoded db name and collection names.

File: documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_response_validation.py (hunk 7 above)
Comment (PatersonProjects) [reply documentdb_tests/compatibility/tests/system/diagnostic/commands/connectionStatus/test_connectionStatus_response_validation.py]: Removed hardcoded names

**Review by eerxuan — APPROVED** [review]

**Review by eerxuan — APPROVED** [review]
